crates/router/src/core/payments.rs

```diff
@@ -4445,7 +4476,43 @@ where
     };
     Ok(connector)
 }
+#[cfg(feature = "v2")]
+#[allow(clippy::too_many_arguments)]
+pub async fn get_connector_choice_for_sdk_session_token<F, Req, D>(
```

Comment:
Is that possible to move these logic inside  `op.to_domain()?.get_connector` function itself?